spec/granite/form/model/validations/nested_spec.rb

```diff
@@ -114,35 +114,75 @@
 
   context 'object field is invalid and referenced object does not include AutosaveAssociation' do
     before do
-      stub_class(:validated_object) do
-        include Granite::Form::Model::Validations
-        include Granite::Form::Model::Attributes
-        include Granite::Form::Model::Primary
-
-        primary :id
+      stub_model(:validated_object) do
         attribute :title, String
         validates_presence_of :title
       end
 
       Main.class_eval do
-        include Granite::Form::Model::Associations
         attribute :object, Object
         validates :object, nested: true
       end
     end
 
-    subject(:instance) { Main.instantiate name: 'hello', object: object, validated_one: {name: 'name'} }
+    subject(:instance) { Main.instantiate name: 'hello', object: object }
+
+    let(:validation_options) { {} }
 
-    context do
+    context 'nested object is valid' do
       let(:object) { ValidatedObject.new(title: 'Mr.') }
+
       it { is_expected.to be_valid }
     end
 
-    context do
+    context 'nested object is invalid' do
       let(:object) { ValidatedObject.new }
+
       it do
         expect { subject.valid? }.not_to raise_error
         expect(subject).not_to be_valid
+        expect(subject.errors.count).to eq(1)
+      end
+
+      context 'nested validation runs twice' do
+        before do
+          Main.class_eval do
+            validates :object, nested: true # Adds existing validation second time
```

Comment:
I doubt only in this place, probably it's better not to use existing `object field is invalid and referenced object does not include AutosaveAssociation` context and use new one and define `validates :object, nested: true` twice within it to make it abvious?

Reply:
I think most of this file needs refactoring, to me the part that you wrote is pretty clear `'nested validation runs twice'` makes me look how it's run twice and I see it's only called once in `before` so I assume this is the second call.

Reply:
I've replaced patching `Main` with stubbing it again instead, so that it's more obvious what subject we have in test.